File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/app/TestKoraApp.java
package ru.tinkoff.kora.database.common.annotation.processor.app;

import org.mockito.Mockito;
import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.jdbc.JdbcConnectionFactory;
import ru.tinkoff.kora.database.jdbc.JdbcRepository;

@KoraApp
public interface TestKoraApp {
    @Repository
    interface TestRepository extends JdbcRepository {
        @Query("INSERT INTO table(value) VALUES (:value)")
        void abstractMethod(String value);
    }

    default JdbcConnectionFactory jdbcQueryExecutorAccessor() {
        return Mockito.mock(JdbcConnectionFactory.class);
    }

    @Root
    default Object mockLifecycle(TestRepository testRepository) {
        return new Object();
    }

}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/app/TestKoraApp.kt
package ru.tinkoff.kora.database.symbol.processor.app

import org.mockito.Mockito
import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.jdbc.JdbcConnectionFactory
import ru.tinkoff.kora.database.jdbc.JdbcRepository

@KoraApp
interface TestKoraApp {
    @Repository
    interface TestRepository : JdbcRepository {
        @Query("INSERT INTO table(value) VALUES (:value)")
        fun abstractMethod(value: String?)
    }

    fun jdbcQueryExecutorAccessor(): JdbcConnectionFactory {
        return Mockito.mock<JdbcConnectionFactory>(JdbcConnectionFactory::class.java)
    }

    @Root
    fun mockLifecycle(testRepository: TestRepository) = Any()
}
